src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -2281,5 +2281,723 @@ public class D { }
+                );
+        }
+
+        [Fact]
+        public void Relational_SignedEnumComplete()
+        {
+            foreach (var typeName in new[] { "sbyte", "short", "int", "long" })
+            {
+                var source = @"
+using System;
+enum E : " + typeName + @"
+{
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven
+}
+class C
+{
+    static void Main()
+    {
+        Console.WriteLine(M(E.Zero));
+        Console.WriteLine(M(E.One));
+        Console.WriteLine(M(E.Two));
+        Console.WriteLine(M(E.Three));
+        Console.WriteLine(M(E.Four));
+        Console.WriteLine(M(E.Five));
+        Console.WriteLine(M(E.Six));
+        Console.WriteLine(M(E.Seven));
+        Console.WriteLine(M((E)100));
+        Console.WriteLine(M((E)(-100)));
+    }
+    static int M(E c) => c switch
+    {
+        >= E.Five => 1,
+        E.Four => 2,
+        E.Three => 3,
+        E.Two => 4,
+        E.One => 5,
+        E.Zero => 6,
+        _ => 7, // handles negative values
+    };
+}";
+                var expectedOutput = @"6
+5
+4
+3
+2
+1
+1
+1
+1
+7";
+                var compilation = CreateCompilation(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+                compilation.VerifyDiagnostics(
+                    );
+                var comp = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            }
+        }
+
+        [Fact]
+        public void Relational_UnsignedEnumComplete()
+        {
+            foreach (var typeName in new[] { "byte", "ushort", "uint", "ulong" })
+            {
+                var source = @"
+using System;
+enum E : " + typeName + @"
+{
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven
+}
+class C
+{
+    static void Main()
+    {
+        Console.WriteLine(M(E.Zero));
+        Console.WriteLine(M(E.One));
+        Console.WriteLine(M(E.Two));
+        Console.WriteLine(M(E.Three));
+        Console.WriteLine(M(E.Four));
+        Console.WriteLine(M(E.Five));
+        Console.WriteLine(M(E.Six));
+        Console.WriteLine(M(E.Seven));
+        Console.WriteLine(M((E)100));
+    }
+    static int M(E c) => c switch
+    {
+        > E.Five => 1,
+        E.Four => 2,
+        E.Three => 3,
+        E.Two => 4,
+        E.One => 5,
+        E.Zero => 6,
+        _ => 7, // handles E.Five
+    };
+}";
+                var expectedOutput = @"6
+5
+4
+3
+2
+7
+1
+1
+1";
+                var compilation = CreateCompilation(source, options: TestOptions.DebugExe, parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Preview));
+                compilation.VerifyDiagnostics(
+                    );
+                var comp = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            }
+        }
+
+        [Fact]
+        public void Relational_SignedEnumExhaustive()
```

Comment:
>Signed [](start = 31, length = 6)

"Unsigned"

Reply:
Will address in the next PR.

---
In reply to: [383372614](https://github.com/dotnet/roslyn/pull/41503#discussion_r383372614) [](ancestors = 383372614)